imblearn/under_sampling/one_sided_selection.py

```diff
@@ -182,6 +182,7 @@ def _sample(self, X, y):
         # If we need to offer support for the indices
         if self.return_indices:
             idx_under = np.flatnonzero(y == self.min_c_)
+        idx_maj = np.flatnonzero(y == self.maj_c_)
```

Comment:
This will make the algorithm failed in the case of multiclass.
`idx_maj` needs to be defined for each class in the `for` loop.

Subsequently, the rest of the algorithm needs to take it into account.